system/lib/webgpu/webgpu.cpp

```diff
@@ -27,3 +38,24 @@ WGPUInstance wgpuCreateInstance(const WGPUInstanceDescriptor* descriptor) {
 
 void wgpuInstanceReference(WGPUInstance) { /* no-op for now */ }
 void wgpuInstanceRelease(WGPUInstance) { /* no-op for now */ }
+
+// WGPUSurface
+
+void wgpuSurfaceGetCapabilities(WGPUSurface surface,
+                                WGPUAdapter adapter,
+                                WGPUSurfaceCapabilities* capabilities) {
+  WGPUTextureFormat preferredFormat =
+      wgpuSurfaceGetPreferredFormat(surface, adapter);
+  assert(preferredFormat == WGPUTextureFormat_BGRA8Unorm ||
+         preferredFormat == WGPUTextureFormat_RGBA8Unorm);
+  capabilities->formatCount = 3;
+  if (preferredFormat == WGPUTextureFormat_RGBA8Unorm) {
+    capabilities->formats = reinterpret_cast<const WGPUTextureFormat*>(
```

Comment:
We should try to avoid reinterpret_cast. Why is it needed? I think `.data()` should already have the right type.